tests/test_read.py

```diff
@@ -90,3 +92,142 @@ def test_read_transformed_records_yields_parsed_dictionary(timdex_dataset_multi_
+    timdex_dataset_with_runs,
+):
+    """This is a somewhat complex test, but demonstrates that only 'current' records
+    are yielded when .load(current_records=True) is applied.
+
+    Given these runs from the fixture:
+    [
+        ...
+        (25, "alma", "2025-01-03", "daily", "index", "run-5"),   <---- filtered to
+        (10, "alma", "2025-01-04", "daily", "delete", "run-6"),  <---- influences current
+        ...
+    ]
+
+    Though we are filtering to run-5, which has 25 total records to-index, we see only 15
+    records yielded.  Why?  This is because while we have filtered to only yield from
+    run-5, run-6 had 10 deletes which made records alma:0|9 no longer "current" in run-5.
+    As we yielded records reverse chronologically, the deletes from run-6 (alma:0-alma:9)
+    "influenced" what records we would see as we continue backwards in time.
+    """
+    # with current_records=False, we get all 25 records from run-5
+    timdex_dataset_with_runs.load(current_records=False, source="alma")
+    df = timdex_dataset_with_runs.read_dataframe(run_id="run-5")
+    assert len(df) == 25
+
+    # with current_records=True, we only get 15 records from run-5
+    # because newer run-6 influenced what records are current for older run-5
+    timdex_dataset_with_runs.load(current_records=True, source="alma")
+    df = timdex_dataset_with_runs.read_dataframe(run_id="run-5")
+    assert len(df) == 15
+    assert list(df.timdex_record_id) == [
+        "alma:10",
+        "alma:11",
+        "alma:12",
+        "alma:13",
+        "alma:14",
+        "alma:15",
+        "alma:16",
+        "alma:17",
+        "alma:18",
+        "alma:19",
+        "alma:20",
+        "alma:21",
+        "alma:22",
+        "alma:23",
+        "alma:24",
+    ]
+
+
+@pytest.mark.skip(reason="All tests for 'current' records will be reworked.")
+def test_dataset_load_current_records_gets_correct_same_day_full_run(
+    timdex_dataset_same_day_runs,
+):
+    """Two full runs were performed on the same day, but 'run-2' was performed most
+    recently.  current_records=True should discover the more recent of the two 'run-2',
+    not 'run-1'."""
+    timdex_dataset_same_day_runs.load(current_records=True, run_type="full")
+    df = timdex_dataset_same_day_runs.read_dataframe()
+
+    assert list(df.run_id.unique()) == ["run-2"]
+
+
+@pytest.mark.skip(reason="All tests for 'current' records will be reworked.")
+def test_dataset_load_current_records_gets_correct_same_day_daily_runs_ordering(
+    timdex_dataset_same_day_runs,
+):
+    """Two runs were performed on 2025-01-02, but the most recent records should be from
+    run 'run-5' which are action='delete', not 'run-4' with action='index'."""
+    timdex_dataset_same_day_runs.load(current_records=True, run_type="daily")
+    first_record = next(timdex_dataset_same_day_runs.read_dicts_iter())
+
+    assert first_record["run_id"] == "run-5"
+    assert first_record["action"] == "delete"
```

Comment:
These were **moved** from `tests/test_dataset.py`, where they probably should have been in the `test_read.py` file this whole time.  

They will get reworked in [TIMX-529](https://mitlibraries.atlassian.net/browse/TIMX-529), maybe significantly, but _what_ they test -- mostly around current records -- will remain the same.